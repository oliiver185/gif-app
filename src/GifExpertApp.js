import React, { useState } from 'react'
import AddCategory from './components/AddCategory';
import { GifGrid } from './components/GifGrid';



export const GifExpertApp = ({defaultCategories= []}) => {

    

const [categories, setCategories] = useState(defaultCategories);
   
    
    
    return (
    <>
        <div className="container">
             <h2 className="titulo">GifExpertApp<br/><span className="titulo-dos">Look for your favorite Gif on our website!</span></h2>
             {/* <h2 className="titulo-dos"></h2> */}
            
            <AddCategory setCategories= {setCategories} />

       </div>
          
        
           

        <ol className="grid-body">
            {
                categories.map( category =>
                <GifGrid key={category} category= {category} />              
                )
            }
        </ol>

        <footer>
            <div className="footer-container">
               
                <p> &copy; 2021 Oliver Petzey</p>
           
            </div>
        </footer>

    </>
    )
}



// export default GifExpertApp;