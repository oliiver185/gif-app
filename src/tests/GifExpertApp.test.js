import React from 'react'
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifExpertApp } from '../GifExpertApp';



describe(' Working with <GifExpertApp/> ', () => {

    test('should work correctly', () => {
        

        const wrapper = shallow(<GifExpertApp/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('should show a list of categories', () => {
        
        const categories = ['One Punch', 'Dragon Ball']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
    
    
})