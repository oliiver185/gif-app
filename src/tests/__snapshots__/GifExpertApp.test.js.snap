// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Working with <GifExpertApp/>  should show a list of categories 1`] = `
<Fragment>
  <div
    className="container"
  >
    <h2
      className="titulo"
    >
      GifExpertApp
      <br />
      <span
        className="titulo-dos"
      >
        Look for your favorite Gif on our website!
      </span>
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
  </div>
  <ol
    className="grid-body"
  >
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
  <footer>
    <div
      className="footer-container"
    >
      <p>
         © 2021 Oliver Petzey
      </p>
    </div>
  </footer>
</Fragment>
`;

exports[` Working with <GifExpertApp/>  should work correctly 1`] = `
<Fragment>
  <div
    className="container"
  >
    <h2
      className="titulo"
    >
      GifExpertApp
      <br />
      <span
        className="titulo-dos"
      >
        Look for your favorite Gif on our website!
      </span>
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
  </div>
  <ol
    className="grid-body"
  />
  <footer>
    <div
      className="footer-container"
    >
      <p>
         © 2021 Oliver Petzey
      </p>
    </div>
  </footer>
</Fragment>
`;
